activerecord/test/cases/connection_pool_test.rb

```diff
@@ -507,7 +507,7 @@ def test_pool_sets_connection_schema_cache
         pool.schema_cache = schema_cache
 
         pool.with_connection do |conn|
-          assert_not_same pool.schema_cache, conn.schema_cache
+          assert_same pool.schema_cache, conn.schema_cache
           assert_equal pool.schema_cache.size, conn.schema_cache.size
           assert_same pool.schema_cache.columns(:posts), conn.schema_cache.columns(:posts)
```

Comment:
I think we can remove these two assertions.  The `assert_same` we added above should be sufficient.